### DISCUSSION TITLE
not a bug report, probably my bad.

### DISCUSSION THREAD

**Discussion by MattPChoy** [chris-greening/instascrape#85] [Q&A]
Hi, trying to scrape and I keep getting this warning: 
```
MissingCookiesWarning: Request header does not contain cookies! It's recommended you pass at least a valid sessionid otherwise Instagram will likely redirect you to their login page.
  warnings.warn(
```
Despite having a `headers` variable:
```
headers = {
    "user-agent": "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Mobile Safari/537.36 Edg/87.0.664.57",
    "cookie": "sessionid=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx;"
}
```
And using it whenever I use the `*.scrape()` method

**Comment by tarob0ba** [comment]
I'll take a look at this.

**Reply by tarob0ba** [reply]
Strange- it shouldn't (in practice) be raising a `KeyError`, since the "cookie" key is present in the dictionary. Even stranger, when I copy the problematic code and make a quick script to test it, it functions properly.

**Reply by MattPChoy** [reply]
@beckettnormington  Thanks for replying, how can I aid in this debugging process? Super weird error!

**Reply by MattPChoy** [reply]
Also, just clarification, when I scrape using the `*.scrape(headers=headers)` method, it gets an error. However, when I scrape using `*.scrape()` it works for a bit (before Instagram redirects me to the login page)

**Reply by tarob0ba** [reply]
Seems like it might be relating to hard coding your session ID. While this shouldn’t be happening, you might have to use an f-string with a variable. Sorry for the suboptimal solution and I’m going to look into this further.

ping: @chris-greening 

Hacky example (copy and pasted from #78):

```python
from instascrape import *

session_id = '' # Your session ID - http://valvepress.com/how-to-get-instagram-session-cookie/

google = Profile('https://www.instagram.com/google/')

headers = {"user-agent": "Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Mobile Safari/537.36 Edg/87.0.664.57",
           "cookie": f"sessionid={session_id};"}

google.scrape(headers=headers)

print(google.followers)
```

**Comment by MattPChoy** [comment]
@chrisgreening
Thanks mate for the reply. Not sure if you'll get this but we'll see what
happens.
I've found that using Selenium resolved this error and I don't have to
worry about session ids changing.
I've encountered the following error though (Error #86)
